components/calendar/src/any_calendar.rs

```diff
@@ -748,13 +762,13 @@ impl AnyCalendar {
     ///
     /// [📚 Help choosing a constructor](icu_provider::constructors)
     #[cfg(feature = "compiled_data")]
-    pub fn new_for_locale(locale: &DataLocale) -> Self {
-        let kind = AnyCalendarKind::from_data_locale_with_fallback(locale);
+    pub fn new_for_locale(prefs: AnyCalendarPreferences) -> Self {
```

Comment:
Should I rename this function?

Reply:
I think so - just `new`?

Reply:
@Manishearth What do you think about the name of this function? (a constructor for AnyCalendar)

Reply:
We already have

```rust
    #[cfg(feature = "compiled_data")]
    pub const fn new(kind: AnyCalendarKind) -> Self
```

Reply:
I'm fine with that being renamed if we need. `new_for_kind(kind)` vs `new(prefs)` or `new(kind)` vs `new_for_prefs(prefs)`/`new_for_locale(prefs)`

Reply:
I made https://github.com/unicode-org/icu4x/issues/5843 to discuss this before 2.0 (but after 2.0 Beta) and unblock this PR